src/modules/scene/question_scene.py

```diff
@@ -19,22 +19,92 @@ def __init__(self, screen, player_state, network):
             .add_solution(MultipleChoiceSolutionBuilder().add_solution("Huang").build())
             .build()
         ])
+        self.__player_state = player_state
 
     def start_scene(self):
         # TODO: Who is responsible to render the next question
-        idx = 0
+        q_idx = 0
         questions = self.get_player_state().get_questions()
+        # TODO: change questions
+        options = questions[q_idx].get_options()
+        option_boxes, optons_box_borders = self.__create_option_boxes(options)
+        # TODO: add a box that encloses the option boxes
+        box_colors = ["red", "green", "blue", "yellow", "purple"]
+        print(options)
 
         while True:
-            if len(questions) == idx:
+            if len(questions) == q_idx:
                 return SceneState.QUIT
 
-            for event in pygame.event.get():
-                if event.type == pygame.QUIT:
-                    pygame.quit()
-                    sys.exit(0)
+            for event in pg.event.get():
+                match event.type:
+                    case pg.QUIT:
+                        pg.quit()
+                        sys.exit(0)
+
+                    case pg.MOUSEBUTTONDOWN:
+                        active = [box.collidepoint(event.pos)
+                                  for box in option_boxes]
+                        for option, clicked in zip(options, active):
+                            if clicked:
+                                print(f"You have selected \"{option}\"!")
+                                # TODO: add click visual effect (change to a brighter/dimmer color)
+                                # TODO: verify solution
+                                # TODO: update player state and current scene state vars
+                                # q_idx += 1
+                                # update options
+                                # TODO: maybe add a next_question method?
+                                options = questions[q_idx].get_options()
 
             self.get_screen().fill("white")
-            questions[idx].draw(self.get_screen())
-            pygame.display.flip()
-            # idx += 1
+            questions[q_idx].draw(self.get_screen())
+
+            for i, (option, box) in enumerate(zip(options, option_boxes)):
+                # print(f"Drawing box with center {box.center}")
+                # TESTING
+                pg.draw.rect(self.get_screen(),
+                             box_colors[i % len(box_colors)], box)
+                # TODO: center text
+                text_surface = STYLE["font"]["text"].render(
+                    option, True, (0, 0, 0))
+                self.get_screen().blit(text_surface, box.center)
+
+            pg.display.flip()
+
+    def __create_option_boxes(self, options):
+        option_boxes, optons_box_borders = [], []
```

Comment:
I don't think we ever used `options_box_borders` in the main loop.

Reply:
Yeah, it's currently a placeholder. I have it as one of my TODOs